bundles/org.openhab.binding.intellicenter2/src/main/resources/OH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,150 @@
+			</channel>
+			<channel id="target-temperature" typeId="temperature">
+				<label>Target temperature</label>
+				<description>The target pool temperature.</description>
+			</channel>
+			<channel id="heater-status" typeId="heater-status">
+				<label>Heater status</label>
+				<description>On if the heater is running, off otherwise.</description>
+			</channel>
+		</channels>
+
+		<representation-property>SNAME</representation-property>
+
+	</thing-type>
+
+	<thing-type id="feature">
+		<supported-bridge-type-refs>
+			<bridge-type-ref id="bridge"/>
+		</supported-bridge-type-refs>
+
+		<label>IntelliCenter2 Feature</label>
+		<description>Feature for IntelliCenter2 Binding</description>
+
+		<channels>
+			<channel id="feature-on-off" typeId="feature-on-off">
+				<label>Feature Status</label>
+				<description>Feature On/Off</description>
+			</channel>
+		</channels>
+
+		<representation-property>OBJNAM</representation-property>
+
+	</thing-type>
+
+	<thing-type id="light">
+		<supported-bridge-type-refs>
+			<bridge-type-ref id="bridge"/>
+		</supported-bridge-type-refs>
+
+		<label>IntelliCenter2 Light</label>
+		<description>Light for IntelliCenter2 Binding</description>
+
+		<channels>
+			<channel id="light-color" typeId="system.color"/>
+			<channel id="light-power" typeId="system.power"/>
+		</channels>
+
+		<representation-property>OBJNAM</representation-property>
+
+	</thing-type>
+
+	<thing-type id="pump">
+		<supported-bridge-type-refs>
+			<bridge-type-ref id="bridge"/>
+		</supported-bridge-type-refs>
+
+		<label>IntelliCenter2 Pump</label>
+		<description>Pump for IntelliCenter2 Binding</description>
+
+		<channels>
+			<channel id="pump-gpm" typeId="pump-gpm"/>
+			<channel id="pump-rpm" typeId="pump-rpm"/>
+			<channel id="pump-power" typeId="pump-power"/>
+		</channels>
+
+		<representation-property>OBJNAM</representation-property>
+
+	</thing-type>
+
+	<channel-type id="temperature">
+		<item-type>Number:Temperature</item-type>
+		<label>Temperature</label>
+		<description>The temperature value.</description>
+		<state readOnly="true"/>
+	</channel-type>
+
+	<channel-type id="heater-status">
+		<item-type>Switch</item-type>
+		<label>Pool Heater</label>
+		<description>ON if the pool heater is running.</description>
+		<state readOnly="true"/>
+	</channel-type>
+
+	<channel-type id="feature-on-off">
+		<item-type>Switch</item-type>
+		<label>Feature Status</label>
+		<description>Feature On/Off</description>
+	</channel-type>
+
+	<channel-type id="pump-gpm">
+		<item-type>Number</item-type>
+		<label>Flow Rate (GPM)</label>
+		<description>Pump Flow Rate (GPM)</description>
+		<state readOnly="true"/>
+	</channel-type>
+
+	<channel-type id="pump-rpm">
+		<item-type>Number</item-type>
+		<label>Pump Speed (RPM)</label>
+		<description>Pump Speed (RPM)</description>
+		<state readOnly="true"/>
+	</channel-type>
+
+	<channel-type id="pump-power">
+		<item-type>Number</item-type>
```

Comment:
Can you be persuaded into implementing UoM? This would then be `Number:Power` and you would need to update the channel with a `QuantityType` with appropriate unit, in this case `Units.WATT`:
```java
new QuantityType<>(n, Units.WATT)
```

Additionally, state below could then be defined like this (with desired number of decimals):
```xml
<state readOnly="true" pattern="%.0f %unit%"/>
```

See https://www.openhab.org/docs/concepts/units-of-measurement.html

Reply:
Updated! Great suggestion.